WalletWasabi.Gui/MainWindow.xaml.cs

```diff
@@ -78,74 +87,114 @@ private async void MainWindow_ClosingAsync(object? sender, CancelEventArgs e)
 			}
 		}
 
-		private async Task ClosingAsync()
+		/// <summary>
+		/// This will try to close the main window of the application.
+		/// </summary>
+		/// <param name="tryToDequeue">If true then the user will get a dialog where a status of the dequing process is shown and the window will be kept open until it successfully finishes. If false then the dequeing process will be handled later at the disposal of the business logic.</param>
+		/// <returns></returns>
+		public async Task ClosingAsync(bool tryToDequeue)
```

Comment:
What if this was `private`? Then it would mean that you cannot call the method from wrong thread. How would you call it then from Program.cs? You would attempt to call something like `Dispatcher.UIThread.InvokeAsync(() => window.Close().ConfigureAwait(false));` instead of `await window.ClosingAsync(tryToDequeue: false).ConfigureAwait(false);`

This is just an idea. But maybe worth exploring.

Reply:
I got your point here. I tried to change this but ended up with unresolvable problems with async or blocking at the `ShowDialogAsync` function. This will be replaced by the new UI so it is not worth to spend more time on this.